proposals/0311-task-locals.md

```diff
@@ -5,54 +5,58 @@
+  + [Go: explicit context passing all the way](#go--explicit-context-passing-all-the-way)
 * [Alternatives Considered](#alternatives-considered)
-  * [Surface API: Key-less value definitions](#surface-api-key-less-value-definitions)
+  + [Surface API: Type-based key definitions](#surface-api--type-based-key-definitions)
+  + [Surface API: Key-less value definitions](#surface-api--key-less-value-definitions)
 * [Rejected Alternatives](#rejected-alternatives)
-  * [Plain-old Thread Local variables](#plain-old-thread-local-variables)
-  * [Dispatch Queue Specific Values](#dispatch-queue-specific-values)
+  + [Plain-old Thread Local variables](#plain-old-thread-local-variables)
+  + [Dispatch Queue Specific Values](#dispatch-queue-specific-values)
 * [Intended use-cases](#intended-use-cases)
-  * [Use case: Distributed Tracing & Contextual Logging](#use-case-distributed-tracing--contextual-logging)
-    * [Contextual Logging](#contextual-logging)
-    * [Function Tracing](#function-tracing)
-    * [Distributed Tracing](#distributed-tracing)
-    * [Future direction: Function wrapper interaction](#future-direction-function-wrapper-interaction)
-  * [Use case: Mocking internals (Swift System)](#use-case-mocking-internals-swift-system)
-  * [Use case: Progress Monitoring](#use-case-progress-monitoring)
-  * [Use case: Executor configuration](#use-case-executor-configuration)
+  + [Use case: Distributed Tracing & Contextual Logging](#use-case--distributed-tracing---contextual-logging)
+    - [Contextual Logging](#contextual-logging)
+    - [Function Tracing](#function-tracing)
+    - [Distributed Tracing](#distributed-tracing)
+    - [Future direction: Function wrapper interaction](#future-direction--function-wrapper-interaction)
+  + [Use case: Mocking internals (Swift System)](#use-case--mocking-internals--swift-system-)
+  + [Use case: Progress Monitoring](#use-case--progress-monitoring)
+  + [Use case: Executor configuration](#use-case--executor-configuration)
 * [Future Directions](#future-directions)
-  * [Tracing annotations with Function Wrappers](#tracing-annotations-with-function-wrappers)
-  * [Specialized TaskLocal Value Inheritance Semantics](#specialized-tasklocal-value-inheritance-semantics)
-    * ["Never" task-local value inheritance](#never-task-local-value-inheritance)
-    * ["always, best effort" task-local value inheritance](#always-best-effort-task-local-value-inheritance)
+  + [Additional configuration options for `@TaskLocal`](#additional-configuration-options-for---tasklocal-)
+  + [Tracing annotations with Function Wrappers](#tracing-annotations-with-function-wrappers)
+  + [Language features to avoid nesting with `withValue`](#language-features-to-avoid-nesting-with--withvalue-)
+  + [Specialized TaskLocal Value Inheritance Semantics](#specialized-tasklocal-value-inheritance-semantics)
+    - ["Never" task-local value inheritance](#-never--task-local-value-inheritance)
```

Comment:
```suggestion
  + [Additional configuration options for `@TaskLocal`](#additional-configuration-options-for-tasklocal)
  + [Tracing annotations with Function Wrappers](#tracing-annotations-with-function-wrappers)
  + [Language features to avoid nesting with `withValue`](#language-features-to-avoid-nesting-with-withvalue)
  + [Specialized TaskLocal Value Inheritance Semantics](#specialized-tasklocal-value-inheritance-semantics)
    - ["Never" task-local value inheritance](#never-task-local-value-inheritance)
```